### PR TITLE
[kjobctl] Move dev dependencies to tools.

### PR DESCRIPTION
**Pull request by mbobrovskyi**
#### What type of PR is this?
/kind cleanup

#### What this PR does / why we need it:
Move dev dependencies to tools on kjobctl.

#### Which issue(s) this PR fixes:

Fixes #

#### Does this PR introduce a user-facing change?

```release-note
NONE
```

### CODE REVIEW

**Review by trasc — COMMENTED** [review]
/lgtm
/approve

File: cmd/experimental/kjobctl/hack/tools/go.mod [hunk 1]
```diff
@@ -0,0 +1,67 @@
+module sigs.k8s.io/kueue/cmd/experimental/kjobctl/hack/tools
```
Comment (alculquicondor) [inline cmd/experimental/kjobctl/hack/tools/go.mod:1]: I think we need to add a rule for this module in `.github/dependabot.yml`

Comment (mbobrovskyi) [reply cmd/experimental/kjobctl/hack/tools/go.mod:1]: #2450

**Pull request merged**